src/quadexpr.jl

```diff
@@ -105,67 +138,27 @@ end
 
 
 ##########################################################################
-# GenericQuadConstraint
-# ∑qᵢⱼ xᵢⱼ  +  ∑ aᵢ xᵢ  +  c  [≤,≥]  0
-# As RHS is implicitly taken to be zero, we store only LHS and sense
-type GenericQuadConstraint{QuadType} <: AbstractConstraint
-    terms::QuadType
-    sense::Symbol
-end
-Base.copy{CON<:GenericQuadConstraint}(c::CON, new_model::Model) = CON(copy(c.terms, new_model), c.sense)
+# TODO: GenericQuadExprConstraint
 
+struct QuadExprConstraint{S <: MOI.AbstractScalarSet} <: AbstractConstraint
+    func::QuadExpr
+    set::S
+end
 
-# Alias for (Float64, Variable)
-const QuadConstraint = GenericQuadConstraint{QuadExpr}
+"""
+    addconstraint(m::Model, c::QuadExprConstraint)
 
-function Base.copy(c::QuadConstraint, new_model::Model)
-    return QuadConstraint(copy(c.terms, new_model), c.sense)
+Add a `QuadExpr` constraint to `Model m`.
+"""
+function addconstraint(m::Model, c::QuadExprConstraint)
+    @assert !m.solverinstanceattached # TODO
```

Comment:
This will be addressed when problem modification support is added?

Reply:
Definitely